Source/ASTextNode2.mm

```diff
@@ -323,10 +320,17 @@ - (NSArray *)exclusionPaths
   return _textContainer.exclusionPaths;
 }
 
-- (void)prepareAttributedString:(NSMutableAttributedString *)attributedString
+- (NSAttributedString *)l_preparedAttributedText
 {
-  ASLockScopeSelf();
```

Comment:
Pretty sure the lock should stay here!

Reply:
Oops, meant to replace that with a lock assertion. I used the `l_` prefix to indicate the method is called while the lock is held. That may be too easy to miss, so I could change it to `locked_` although locked assertions will probably prevent people from missing it in reality (unit tests).